File: python-examples/examples/basic_chat.py
import os
import sys
from openai import OpenAI

from dotenv import load_dotenv

load_dotenv()


class ChatBot:
    def __init__(self, system_prompt: str = ""):
        self.client = OpenAI(
            base_url=os.getenv("LLM_BASE_URL", "https://api.llmshare.network/v1"),
            api_key=os.getenv("LLM_API_KEY", ""),
        )
        self.model = os.getenv("LLM_MODEL", "gpt-3.5-turbo")
        self.messages = []
        
        if system_prompt:
            self.messages.append({
                "role": "system",
                "content": system_prompt
            })
        
        if not self.client.api_key:
            raise ValueError("API Key 未配置，请设置 LLM_API_KEY 环境变量")

    def chat(self, message: str) -> str:
        self.messages.append({
            "role": "user",
            "content": message
        })
        
        try:
            response = self.client.chat.completions.create(
                model=self.model,
                messages=self.messages,
                temperature=0.7,
                max_tokens=2048,
            )
            
            assistant_message = response.choices[0].message.content or ""
            
            self.messages.append({
                "role": "assistant",
                "content": assistant_message
            })
            
            return assistant_message
        except Exception as e:
            print(f"聊天请求失败: {e}", file=sys.stderr)
            raise

    def clear_history(self):
        system_message = None
        for msg in self.messages:
            if msg["role"] == "system":
                system_message = msg
                break
        
        self.messages = [system_message] if system_message else []

    def get_history(self):
        return self.messages.copy()


def main():
    bot = ChatBot(
        system_prompt="你是一个友好、专业的AI助手。"
    )
    
    print("=== Python 聊天机器人演示 ===\n")
    
    messages = [
        "你好，请介绍一下你自己",
        "你擅长什么领域？",
        "谢谢你！"
    ]
    
    for msg in messages:
        print(f"用户: {msg}")
        response = bot.chat(msg)
        print(f"AI: {response}\n")
    
    print(f"对话历史: {len(bot.get_history())} 条消息")


if __name__ == "__main__":
    main()

File: go-examples/examples/basic_chat.go
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/sashabaranov/go-openai"
)

type ChatBot struct {
	client  *openai.Client
	model   string
	messages []openai.ChatCompletionMessage
}

func NewChatBot(systemPrompt string) *ChatBot {
	config := openai.DefaultConfig(os.Getenv("LLM_API_KEY"))
	config.BaseURL = os.Getenv("LLM_BASE_URL")
	if config.BaseURL == "" {
		config.BaseURL = "https://api.llmshare.network/v1"
	}

	client := openai.NewClientWithConfig(config)

	model := os.Getenv("LLM_MODEL")
	if model == "" {
		model = "gpt-3.5-turbo"
	}

	bot := &ChatBot{
		client: client,
		model:  model,
		messages: []openai.ChatCompletionMessage{},
	}

	if systemPrompt != "" {
		bot.messages = append(bot.messages, openai.ChatCompletionMessage{
			Role:    openai.ChatMessageRoleSystem,
			Content: systemPrompt,
		})
	}

	return bot
}

func (bot *ChatBot) Chat(message string) (string, error) {
	bot.messages = append(bot.messages, openai.ChatCompletionMessage{
		Role:    openai.ChatMessageRoleUser,
		Content: message,
	})

	resp, err := bot.client.CreateChatCompletion(
		context.Background(),
		openai.ChatCompletionRequest{
			Model:       bot.model,
			Messages:    bot.messages,
			Temperature: 0.7,
			MaxTokens:   2048,
		},
	)
	if err != nil {
		return "", fmt.Errorf("chat request failed: %w", err)
	}

	assistantMessage := resp.Choices[0].Message.Content

	bot.messages = append(bot.messages, openai.ChatCompletionMessage{
		Role:    openai.ChatMessageRoleAssistant,
		Content: assistantMessage,
	})

	return assistantMessage, nil
}

func (bot *ChatBot) ClearHistory() {
	if len(bot.messages) > 0 && bot.messages[0].Role == openai.ChatMessageRoleSystem {
		bot.messages = bot.messages[:1]
	} else {
		bot.messages = []openai.ChatCompletionMessage{}
	}
}

func (bot *ChatBot) GetHistory() []openai.ChatCompletionMessage {
	return bot.messages
}

func main() {
	apiKey := os.Getenv("LLM_API_KEY")
	if apiKey == "" {
		fmt.Println("Error: LLM_API_KEY environment variable is not set")
		os.Exit(1)
	}

	bot := NewChatBot("你是一个友好、专业的AI助手。")

	fmt.Println("=== Go 聊天机器人演示 ===\n")

	messages := []string{
		"你好，请介绍一下你自己",
		"你擅长什么领域？",
		"谢谢你！",
	}

	for _, msg := range messages {
		fmt.Printf("用户: %s\n", msg)
		response, err := bot.Chat(msg)
		if err != nil {
			fmt.Printf("错误: %v\n", err)
			os.Exit(1)
		}
		fmt.Printf("AI: %s\n\n", response)
	}

	fmt.Printf("对话历史: %d 条消息\n", len(bot.GetHistory()))
}
